pkg/util/k8sutil/tpr_util.go

```diff
@@ -74,16 +78,33 @@ func GetClusterTPRObject(restcli rest.Interface, ns, name string) (*spec.Cluster
 	return readOutCluster(b)
 }
 
-// UpdateClusterTPRObject updates the given TPR object.
-// ResourceVersion of the object MUST be set or update will fail.
-func UpdateClusterTPRObject(restcli rest.Interface, ns string, c *spec.Cluster) (*spec.Cluster, error) {
-	if len(c.Metadata.ResourceVersion) == 0 {
-		return nil, errors.New("k8sutil: resource version is not provided")
-	}
-	return updateClusterTPRObject(restcli, ns, c)
+// AtomicUpdateClusterTPRObject takes an update function which is executed before attempting
+// to update the Cluster TPR object. The entire process is retried
+// again until the update is successful or a non-conflict error is returned.
+func AtomicUpdateClusterTPRObject(restcli rest.Interface, name, namespace string, updateFunc ClusterTPRModifier) (*spec.Cluster, error) {
+	var updatedCluster *spec.Cluster
+	err := retryutil.Retry(5*time.Second, 5, func() (done bool, err error) {
+		currCluster, err := GetClusterTPRObject(restcli, namespace, name)
+		if err != nil {
+			return false, err
+		}
+
+		updateFunc(currCluster)
+
+		updatedCluster, err = UpdateClusterTPRObject(restcli, namespace, currCluster)
+		if err != nil {
+			if apierrors.IsConflict(err) {
```

Comment:
Right.
This is a key path we might want to test. Maybe put the test into "cliente2e/"?
But since you have also verified the code path, I'm fine to write the test PR separately to minimize each PR.